local/src/core/ProxyVerifier.cc

```diff
@@ -1799,8 +1800,25 @@ RuleCheck::options_init()
       static_cast<single_field_function_type>(make_absence);
   options[swoc::TextView(YAML_RULE_CONTAINS)] =
       static_cast<single_field_function_type>(make_contains);
-  options[swoc::TextView(YAML_RULE_PREFIX)] = static_cast<single_field_function_type>(make_prefix);
-  options[swoc::TextView(YAML_RULE_SUFFIX)] = static_cast<single_field_function_type>(make_suffix);
+  options[swoc::TextView(YAML_RULE_PREFIX)] =
+      static_cast<single_field_function_type>(make_prefix);
+  options[swoc::TextView(YAML_RULE_SUFFIX)] =
+      static_cast<single_field_function_type>(make_suffix);
+
+  url_rule_options = URLRuleOptions();
+  using url_function_type = std::shared_ptr<RuleCheck> (*)(YAML_URL, swoc::TextView);
```

Comment:
There shouldn't be a situation where C style function pointers work and `std::function` doesn't.

Reply:
This turned out to be nontrivial. Dropping this is fine with me because this PR didn't introduce this anyway. What we have works, so I don't want to hold up this PR for that.